app/src/main/java/io/element/android/x/node/RoomFlowNode.kt

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2023 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.element.android.x.node
+
+import android.os.Parcelable
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import com.bumble.appyx.core.composable.Children
+import com.bumble.appyx.core.lifecycle.subscribe
+import com.bumble.appyx.core.modality.BuildContext
+import com.bumble.appyx.core.node.Node
+import com.bumble.appyx.core.node.ParentNode
+import com.bumble.appyx.navmodel.backstack.BackStack
+import io.element.android.x.architecture.bindings
+import io.element.android.x.architecture.createNode
+import io.element.android.x.core.di.DaggerComponentOwner
+import io.element.android.x.di.RoomComponent
+import io.element.android.x.features.messages.MessagesNode
+import io.element.android.x.matrix.room.MatrixRoom
+import kotlinx.parcelize.Parcelize
+import timber.log.Timber
+
+class RoomFlowNode(
+    buildContext: BuildContext,
+    private val room: MatrixRoom,
+    private val backstack: BackStack<NavTarget> = BackStack(
+        initialElement = NavTarget.Messages,
+        savedStateMap = buildContext.savedStateMap,
+    ),
+) : ParentNode<RoomFlowNode.NavTarget>(
+    navModel = backstack,
+    buildContext = buildContext
+), DaggerComponentOwner {
+
+    override val daggerComponent: Any by lazy {
+        parent!!.bindings<RoomComponent.ParentBindings>().roomComponentBuilder().room(room).build()
+    }
+
+    init {
+        lifecycle.subscribe(
+            onCreate = { Timber.v("OnCreate") },
+            onDestroy = { Timber.v("OnDestroy") }
+        )
```

Comment:
We have this in all our Nodes?

Reply:
No, I want to add this on important ones